From: Sergio Costas Rodriguez <sergio.costas@canonical.com>
Date: Mon, 16 Jun 2025 16:42:46 +0200
Subject: a11y: Specify what is being selected in avatar

When the user selects to change the user avatar, and the stock
avatars are shown, the screen reader just reads the name of the
first avatar ("Bycicle"), but gives no indication of what is
describing.

This patch adds a group name ("Stock avatar") to the list of
avatars available by default, thus giving context to the screen
reader.

It also adds a role to the avatar buttons in a separate
commit, to simplify backporting separately (the group name
requires a translation, thus is not suitable for backporting).

Origin: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/3183
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-control-center/+bug/2115973
---
 panels/system/users/cc-avatar-chooser.ui | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/panels/system/users/cc-avatar-chooser.ui b/panels/system/users/cc-avatar-chooser.ui
index 0f19c6e..bbdcc80 100644
--- a/panels/system/users/cc-avatar-chooser.ui
+++ b/panels/system/users/cc-avatar-chooser.ui
@@ -13,6 +13,10 @@
             <property name="homogeneous">True</property>
             <property name="max-children-per-line">5</property>
             <property name="column-spacing">10</property>
+            <property name="accessible-role">group</property>
+            <accessibility>
+              <property name="label" translatable="yes">Stock avatars</property>
+            </accessibility>
           </object>
         </child>
         <child>
